src/PostgREST/Middleware.hs

```diff
@@ -42,15 +40,16 @@ runWithClaims conf time app req = do
         if M.null claims && not (null tokenStr)
           then clientErr "Invalid JWT"
           else do
-            let cmdBatch = mconcat $ claimsToSQL claims
-            unless (BS.null cmdBatch) (H.sql cmdBatch)
+            H.sql . mconcat $
+              setRole (
+                fromMaybe anon (M.lookup "role" claims)
```

Comment:
shouldn't this ordering and default setting a responsibility of claimsToSQL ? After all it returns an array of commands to be executed, why not return the array with everything already set properly?